spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java

```diff
@@ -247,7 +260,8 @@ public MultiListenerBean multiListener() {
 
 		@Bean
 		public ProducerFactory<Integer, String> producerFactory() {
-			return new DefaultKafkaProducerFactory<>(producerConfigs());
+			Map<String, Object> configs = producerConfigs();
+			return new DefaultKafkaProducerFactory<>(configs, producerStrategy(configs));
```

Comment:
This configuration smells to make me duplicate `configs`.
Although finally the first one becomes useless (see my comment on `DefaultKafkaProducerFactory`).

So, I insist that we should just end up with a couple of setters on the `Factories` and don't introduce any "buggy" strategies.